minidump-processor/src/processor.rs

```diff
@@ -763,3 +769,107 @@ where
 
     Ok(state)
 }
+
+/// Fix the crash address if a non-canonical access caused a crash
+///
+/// Amd64 has the concept of a "canonical addressing", which requires that the upper 16 bits of
+/// an address contain the same binary digit as bit 48. A violation of this rule triggers a
+/// General Protection Fault instead of the usual Page Fault, which unfortunately means that the
+/// OS has no idea what memory address actually caused the issue
+///
+/// If `exception_info` contains the markers of a non-canonical exception, and it also contains
+/// memory access info from analyzing the CPU instruction with `op_analysis`, this module will
+/// attempt to determine what address the CPU was instructed to access when the GPF occurred
+///
+/// Nothing will be changed if the crash was not caused by a non-canonical access
+fn fix_non_canonical_crash_address(
+    system_info: &SystemInfo,
+    exception_info: &mut crate::ExceptionInfo,
+) {
+    use minidump_common::errors as minidump_errors;
+    use system_info::{Cpu, Os};
+
+    // The range of non-canonical addresses in the current 48-bit implementation
+    // See: https://en.wikipedia.org/wiki/X86-64#Virtual_address_space_details
+    const NON_CANONICAL_RANGE: RangeInclusive<u64> = 0x0000_8000_0000_0000..=0xffff_7fff_ffff_ffff;
+
+    // Only Amd64 has non-canonical addresses
+    if system_info.cpu != Cpu::X86_64 {
+        return;
+    }
+
+    // Different OSes have unique ways of reporting this error
+    match system_info.os {
+        // Windows reports it as EXCEPTION_ACCESS_VIOLATION_READ, address 0xffffffffffffffff
+        Os::Windows => {
+            if exception_info.reason
```

Comment:
These series of `if` statements might be easier to read if you combined them into `match` statements.

Reply:
Fair enough! I attached a new commit - It does seem to read better